maas/data_source_maas_zone.go

```diff
@@ -0,0 +1,49 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceMaasZone() *schema.Resource {
+	return &schema.Resource{
+		Description: "Provides details about an existing MAAS zone.",
+		ReadContext: dataSourceZoneRead,
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "A brief description of the zone.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The zone's name.",
+			},
+		},
+	}
+}
+
+func dataSourceZoneRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*client.Client)
```

Comment:
We recently swapped this out with a clientConfig object, you'll need to swap this out with:
`client := meta.(*ClientConfig).Client`